### ISSUE TITLE
[custom_components.ble_monitor] Invalid data: No encription key found

### ISSUE THREAD

**Issue by djgragra** [custom-components/ble_monitor#296]
I receive this ERROR (Thread-3) [custom_components.ble_monitor] Invalid data: No encription key found
I have 3 LYWSD03MMC and 1 LYWSDCGQ. All the thermometer works fine but in the log I receive the error.
How to solve?

**Comment by Ernst79** [comment]
What firmware do you use? ATC or the original Xiaomi firmware?

**Comment by hung2kgithub** [comment]
LYWSD03MMC on original Xiaomi firmware 0.1.3 still get the same error

**Comment by Ernst79** [comment]
Yes, that release was for another bug.

 In my previous message, I have some question for you, to help me fix this. could you have a look at these questions.

**Comment by hung2kgithub** [comment]
LYWSD03MMC original firmware 0109

**Comment by Ernst79** [comment]
Oh, sorry, you mentioned that in your previous post. My mistake.

**Comment by Ernst79** [comment]
Can you change line 805-807 in `__init__.py` In the folder custom_components/ble_monitor with the following code. 

```
            except KeyError:
                # no encryption key found
                raise NoValidError("No encryption key found for sensor with mac address: " + "".join("{:02X}".format(x) for x in xiaomi_mac_reversed[::-1]))
```

it will still throw the same error, but it will display the MAC address of the sensor that is causing the error. Next, double check the encryption key e.g. with [Xiaomi cloud token extractor](https://github.com/PiotrMachowski/Xiaomi-cloud-tokens-extractor)

**Comment by djgragra** [comment]
I use the original firmware

**Comment by Ernst79** [comment]
Ah, sorry, it didn’t see that a different person responded. Please check my previous post. Let me know if the sensor that is causing the error is in your config (check the MAC address) with the modified code and please double check the encryption key. I understand that the sensor works fine, but I need to understand what is wrong to fix it.

**Comment by djgragra** [comment]
> Ah, sorry, it didn’t see that a different person responded. Please check my previous post. Let me know if the sensor that is causing the error is in your config (check the MAC address) with the modified code and please double check the encryption key. I understand that the sensor works fine, but I need to understand what is wrong to fix it.


I change le code  in __init__.py as you suggested and I have the 3 LYWSD03MMC sensors that generate the error.
This morning I recovered the binkey again but the problem remains even though they are actually working fine and reporting temperature, humidity and battery.
To connect the sensors I use two esp32 m5stick-c via esphome. The firmware of the sensors is original, to recover the bindkey I use https://atc1441.github.io/TelinkFlasher.html

2021-02-28 13:11:43 ERROR (Thread-3) [custom_components.ble_monitor] Invalid data: No encryption key found for sensor with mac address: A4C138CBA3D6
2021-02-28 13:11:43 ERROR (Thread-3) [custom_components.ble_monitor] Invalid data: No encryption key found for sensor with mac address: A4C138F6855D
2021-02-28 13:12:01 ERROR (Thread-3) [custom_components.ble_monitor] Invalid data: No encryption key found for sensor with mac address: A4C138B38869
2021-02-28 13:12:43 ERROR (Thread-3) [custom_components.ble_monitor] Invalid data: No encryption key found for sensor with mac address: A4C138F6855D
2021-02-28 13:13:43 ERROR (Thread-3) [custom_components.ble_monitor] Invalid data: No encryption key found for sensor with mac address: A4C138CBA3D6
2021-02-28 13:14:44 ERROR (Thread-3) [custom_components.ble_monitor] Invalid data: No encryption key found for sensor with mac address: A4C138CBA3D6

**Comment by Ernst79** [comment]
> To connect the sensors I use two esp32 m5stick-c via esphome

You say you connect to these sensors with ESPhome. So, I guess you have connected these sensors to the ESPHome integration, and not to BLE monitor (which is a different integration). To explain the difference, ESPHome connects via the Bluetooth of the esp32 devices, BLE monitor uses the bluetooth of the machine HA is running on (e.g. Raspberry Pi). 

Can you check if there is an encryption key in the options menu of the BLE Monitor integration?

**Comment by djgragra** [comment]
> > To connect the sensors I use two esp32 m5stick-c via esphome
> 
> You say you connect to these sensors with ESPhome. So, I guess you have connected these sensors to the ESPHome integration, and not to BLE monitor (which is a different integration). To explain the difference, ESPHome connects via the Bluetooth of the esp32 devices, BLE monitor uses the bluetooth of the machine HA is running on (e.g. Raspberry Pi).
> 
> Can you check if there is an encryption key in the options menu of the BLE Monitor integration?


The only settings menu I have is the one in the attached image.

I'm probably doing something wrong, using esphome there is no need to unsare
Passive BLE monitor?
![Schermata 2021-02-28 alle 15 04 16](https://user-images.githubusercontent.com/65872226/109421274-ed87e300-79d6-11eb-9ed7-e6398e170080.png)

**Comment by Ernst79** [comment]
You can check by clicking on devices in the bottom. This will open a pulldown menu, where you can select your device when clicking submit, it will open a new window.. If you only use ESPhome to get the readings from all your sensors, there is no need to use BLE monitor. But some people use both, to extend the range. 

I think I will move the error message to debug level in a next update, so it won’t show up.

**Comment by Ernst79** [comment]
Fixed in 1.1.4
